packages/ui/src/utilities/deepMerge.ts

```diff
@@ -23,3 +23,31 @@ export function deepMergeSimple<T = object>(obj1: object, obj2: object): T {
 
   return output as T
 }
+
+/**
+ * Deep merge two objects. Arrays are concatenated, objects are merged recursively.
+ *
+ * @param obj1 base object
+ * @param obj2 object to merge "into" obj1
+ */
+
+export function deepMerge<T = object>(obj1: any, obj2: any): T {
```

Comment:
Elsewhere we `import deepMerge from 'deepmerge'`. I'm not sure if this function works differently than that package.